crates/swc_ecma_utils/src/number.rs

```diff
@@ -49,7 +45,8 @@ impl JsNumber {
             return 0;
         }
 
-        self.0.trunc() as u32
+        // pow(2, 32) = 4294967296
+        self.0.trunc().rem_euclid(4294967296.0) as u32
```

Comment:
I've updated the implementation, and all the tests have passed. 
Take a look to see if we need to add more test cases. @levi-nz

Reply:
lgtm, maybe we could see if v8 or another project has some test cases? https://github.com/v8/v8/blob/main/src/numbers/conversions.h

I think the current test cases are (probably) fine, as long as we're testing for overflowing, the rest should work fine